### PR TITLE
Add Support for Retrieving Custom Pressure Levels and Refactor define_level Function

### PR DESCRIPTION
**Pull request by MclTTI**
### Summary
The `define_level` function has been updated to handle both predefined pressure levels (plev8, plev19, and plev37) and custom user-defined pressure levels specified in the configuration of `ERA5_retrieve_postproc`.

### Changes Made
- **User-defined Levels**: The function now accepts a list of pressure levels from the user, allowing for more flexible configurations.
- **Validation**: User-specified pressure levels can include explicit units (e.g., 'hPa') or be provided without units. These inputs are validated against the available ERA5 pressure levels.
- **Data Structure**: The predefined pressure levels have been relocated to a dictionary named `plevs`, which now includes both predefined and ERA5 pressure levels for improved organization.

### References
A list of valid ERA5 pressure levels can be found here: [ERA5 Data Documentation](https://confluence.ecmwf.int/display/CKB/ERA5%3A+data+documentation).

### CODE REVIEW

File: CDS_retriever.py [hunk 1]
```diff
@@ -132,31 +134,121 @@ def year_retrieve(dataset, var, freq, year, grid, levelout, area, outdir, reques
                 os.remove(f)
 
 
-# define propertes for vertical levels
+
+plevs = {'ERA5' : ['1000', '975', '950', '925',
```
Comment (oloapinivad) [inline CDS_retriever.py:138]: did not try this, but this is dictionary based on the dataset. Did you modified the way this is read? At L246 it does not seem to me you are accessing `levels['ERA5']`... of course I might be wrong!

Comment (MclTTI) [reply CDS_retriever.py:138]: I slightly modified the logic of the function. Your 'define_level' assigns 'level' and 'level kind' based on a series of switches, while mine does the same but also includes a check for custom pressure levels. It verifies whether the request is for predefined levels (stored in the 'plevs' dictionary) or for (one or multiple) custom pressure levels, checking their validity against the 'plevs['ERA5']' list.

The main difference is the validation of arbitrary pressure levels, which I handle at line 199. Additionally, users can input arbitrary levels with or without the unit of measure (hPa only, for now, as per the regex at line 190).

I also generally prefer dictionary searches over using if blocks.

Comment (oloapinivad) [reply CDS_retriever.py:138]: Now I undersand `plevs['ERA5'] is the FULL set of levels for ERA5

**Comment by oloapinivad** [comment]
Merging this

**Pull request merged**